packages/suite/src/reducers/store.ts

```diff
@@ -104,32 +105,38 @@ type RootReducerShape = typeof rootReducer;
 type PreloadedState = Partial<AppState>;
 type InferredAction = Parameters<RootReducerShape>[1];
 
-export const initStore = (
-    preloadStoreAction?: PreloadStoreAction,
-    statePatch?: Record<string, any>,
-) => {
-    // get initial state by calling STORAGE.LOAD action with optional payload
-    // payload will be processed in each reducer explicitly
-    const preloadedState = preloadStoreAction
-        ? rootReducer(undefined, preloadStoreAction)
+export const initStore = <T extends Record<string, any>>(options: {
```

Comment:
I don't quite get this, why are the services handled here, when it seems to be completely independent from what the rest of this function is doing? Why not make a new function? 

And also shouldn't services be passed down to extraDependencies and not the other way around? What if we need to create a service that needs to be used through extraDependencies but also depends on other services?

Reply:
Yeah, this idiotic extra dependencies parameter. Years ago, i tried to tell the maintainers that it sucks, coz you don't have access to store from there.  They said that I need to make my.own middleware for it instead of using official redux thunk. 

You are right that it should be done like so. Since we use the official redux thunk where extra deps don't have access to the store, I'd say that either write own middleware (not a fan) or declare thst eversthjng in exrra dependencies has to be without any other dependency.